internal/pkg/cli/pipeline_init.go

```diff
@@ -450,10 +446,18 @@ func BuildPipelineInitCmd() *cobra.Command {
     --environments "stage,prod" \
     --deploy`,
 		PreRunE: runCmdE(func(cmd *cobra.Command, args []string) error {
+			err := opts.Validate()
+			if err != nil {
+				return err
+			}
+
 			projectEnvs, err := opts.getEnvNames()
 			if err != nil {
 				return fmt.Errorf("couldn't get environments: %w", err)
 			}
+			if len(projectEnvs) == 0 {
```

Comment:
what's bad about leaving this inside the `Validate()` call on ln 449?

Reply:
It will fail because projectEnvs will not be set. Otherwise validate() function cannot be called before opts.getEnvNames()